### DISCUSSION TITLE
128MB N64 ROM?

### DISCUSSION THREAD

**Discussion by PsychoFox11** [sanni/cartreader#773] [Need help]
I'm attempting to dump a homebrew game, it is basically like a repro cart would be with the fpga chip and a rom chip. Looking up the ROM chip, it holds 128MB. The game doesn't seem like it would take that much but maybe it has stuff in that area or the dev just let it get bloated cause memory is cheap. Is this not dumpable with the Sanni? I've attempted at every size the Sanni offers, which is max 64MB. It does recognize the title and the CRC1 value.

**Comment by PsyK0p4T** [comment]
Repro/unofficial PCBs can be dumped for sure, I did it already with some carts which had simple hardware. Just not sure if yours is really different and if the fpga could be annoying... at least if you see the game title, it's a good start !
I can make you a beta version to dump up to 128MB. Are you on our Discord server ? Could be faster to test the behavior with you, and maybe update OSCR code then.

**Reply by PsyK0p4T** [reply]
Here the updated ino file, to be used with current v12.5
[N64.zip](https://github.com/sanni/cartreader/files/11232783/N64.zip)

**Reply by PsychoFox11** [reply]
Yeah I'm in your Discord I just never really use Discord normally but I can jump in there when I get a chance. I do know how to update the N64 file on my own. Do you want me to try that first, or wait until we test something else on Discord first? It will at least be late tonight before I can be on Discord due to work, if not tomorrow.

**Reply by PsyK0p4T** [reply]
Ah cool, I just don't know your nick there, just message me when you'll be around ;)
Sure, you can try to update with file now and let me know the results on Discord. Logs, rom, pictures, all will interesting.
Prepare yourself, because dumping 128 MB will be VERY long...

**Reply by PsychoFox11** [reply]
Just curious, as I'm new to dumping, and am using all the slots on my Sanni with several adapters on the way - is it common for much modern homebrew etc. to not be readable? Not just N64 but I've tried several Genesis games, a couple have worked but many read a checksum of 0000 to start, then return an invalid one and produce a non-working ROM. Just trying to record hashes on these. No luck with NES either but in that case I also don't know the mapper and if I find the chip size I still don't know how much is prg/chr. I imagine there's some anti-dumping techniques in use on modern stuff but am just now getting into this world. I've verified quite a bit in the no-intro database and that is fun for me!
I'll follow up on Discord about this specific one when I can, but was wondering across systems in general.

**Reply by PsyK0p4T** [reply]
Technology evolved a lot since 25-30 years, and aftermarket manufacturers are also trying to push the limits of consoles by using new stuff. They take advantage of bigger chips, embedded hardwares, fpga etc... It's not always for protection reasons. Sometimes it is (like Paprium), but sometimes not really (like Demons of Asteborg).
For the MD/Genesis in particular, new SDKs and custom tools are used to make recent games. Sometimes they don't format the rom header correctly (rom addresses, checksum etc) and we have to add new methods to handle them.

**Reply by PsychoFox11** [reply]
Thanks, yeah I knew Paprium was an issue. I have a copy but didn't bother trying, but it is kinda why I wondered about the others. I have a lot of recent/homebrew I'd love to document, but out of what I've tried so far, only 3 Genesis games, and an N64 game, only one worked, and I can't say for sure about the NES stuff cause I was likely using wrong size values. Happy to add any info I can about any that have issues.

**Reply by PsyK0p4T** [reply]
Hope I can help with Genesis and N64 at least. For NES, nsx will be the one to reach ;)

**Reply by PsychoFox11** [reply]
So I was wrong about being in the discord, and don't see an obvious point of entry - where do I find that? I was just in another talking about Sanni a while back and misremembered.

Testing the file now. I did post an error here earlier but believe I know what I did wrong and have corrected it.

**Reply by PsychoFox11** [reply]
Nevermind, still getting the error.
![sanniN64](https://user-images.githubusercontent.com/5882157/232176016-50ce3955-b2f4-42ed-be14-7e519ae5625e.PNG)

**Reply by PsyK0p4T** [reply]
It has to be used with the current v12.5, you're probably in 12.4 or lower.
- download the current dev version : https://github.com/sanni/cartreader/archive/refs/heads/master.zip
- overwrite the files in your sketchbook with these files
- open the Arduino compiler, and enable your hardware version in "config.h"
- try to flash your cart reader
- if it's all good, overwrite the n64.ino with mine ;)

**Reply by PsyK0p4T** [reply]
And the Discord link : https://discord.gg/YVxv7BnzXt

**Reply by PsychoFox11** [reply]
Thanks. I got the reader updated properly, but the 128MB dump still wasn't any good. No matter the size, if I try in Project64 I get an unknown memory function or something like that... This game is also on GB Color and only 64kb there. The N64 game is basically identical, just more colors and higher res, so I can't imagine it is huge unless they stored raw music or something. The chip is definitely 128MB from what I've searched though.
I'll try to hit you up on discord tomorrow.

**Reply by PsyK0p4T** [reply]
Could be the dump, or could be the emulator too. If the game is an homebrew, developed recently, there is a little chance that it's not supported. Have you tried others, like mupen64plus ?
Sure, message me later.
